logos/benchmarks/run_bench.sh

```
@@ -120,10 +136,17 @@ bench_args=(
 [[ -n "$LOGOS_KEY" ]] && bench_args+=(--logos-key "$LOGOS_KEY")
+# Quick-debug subsetting: SCENARIOS=logos-nosleep PATTERNS=mixed runs just that
+# scenario/pattern. Empty = all scenarios / all 4 patterns.
+[[ -n "$SCENARIOS" ]] && bench_args+=(--scenarios "$SCENARIOS")
+[[ -n "$PATTERNS" ]] && bench_args+=(--patterns "$PATTERNS")
 [[ "$ONLY_OLLAMA" == "1" ]] && bench_args+=(--only-ollama)
 [[ "$MANAGE_CALIB_WINDOW" == "0" ]] && bench_args+=(--no-manage-calibration-window)
 [[ "$SHELLY" == "1" ]] && bench_args+=(--shelly --shelly-port "$SHELLY_PORT" --shelly-transport "$SHELLY_TRANSPORT" --shelly-ingest-image "$SHELLY_INGEST_IMAGE")
-[[ "$RESET_CALIBRATION" == "1" ]] && bench_args+=(--reset-calibration --calibration-provider-ids $CALIBRATION_PROVIDER_IDS)
+# Provider IDs are needed whether or not we reset: without a full reset the run
+# still triggers calibration for any model the worker never calibrated.
+[[ -n "$CALIBRATION_PROVIDER_IDS" && "$SKIP_CALIBRATION" != "1" ]] && bench_args+=(--calibration-provider-ids $CALIBRATION_PROVIDER_IDS)
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)